src/html/scan.js

```diff
@@ -432,7 +455,120 @@ function updateOptions(data) {
+  let disable = false;
+  for (const [b, _v] of Object.entries(buttonActions)) {
+    for (const [p, v] of Object.entries(_v)) {
+      if (v['action'] !== 0 && (_(`select-press-${b}-${p}`).value === '' || _(`select-long-${b}-${p}`).value === '' || _(`select-short-${b}-${p}`).value === '')) {
+        disable = true;
+      }
+    }
+  }
+  _('submit-actions').disabled = disable;
+}
+
+function changeAction(b, p, value) {
+  buttonActions[b][p]['action'] = value;
+  if (value === 0) {
+    _(`select-press-${b}-${p}`).value = '';
+    _(`select-long-${b}-${p}`).value = '';
+    _(`select-short-${b}-${p}`).value = '';
+  }
+  checkEnableButtonActionSave();
+}
+
+function changePress(b, p, value) {
+  buttonActions[b][p]['is-long-press'] = (value==='true');
+  _(`mui-long-${b}-${p}`).style.display = value==='true' ? 'block' : 'none';
+  _(`mui-short-${b}-${p}`).style.display = value==='true' ? 'none' : 'block';
+  checkEnableButtonActionSave();
+}
+
+function changeCount(b, p, value) {
+  buttonActions[b][p]['count'] = parseInt(value);
+  _(`select-long-${b}-${p}`).value = value;
+  _(`select-short-${b}-${p}`).value = value;
+  checkEnableButtonActionSave();
+}
+
+function appendRow(b,p,v) {
+  const row = _('button-actions').insertRow();
+  row.innerHTML = `
+<td>
+  Button ${parseInt(b)+1}
+</td>
+<td>
+  <div class="mui-select">
+    <select onchange="changeAction(${b}, ${p}, parseInt(this.value));">
+      <option value='0' ${v['action']===0 ? 'selected' : ''}>Unused</option>
+      <option value='1' ${v['action']===1 ? 'selected' : ''}>Increase Power</option>
+      <option value='2' ${v['action']===2 ? 'selected' : ''}>Go to VTX Band Menu</option>
+      <option value='3' ${v['action']===3 ? 'selected' : ''}>Go to VTX Channel Menu</option>
+      <option value='4' ${v['action']===4 ? 'selected' : ''}>Send VTX Settings</option>
+      <option value='5' ${v['action']===5 ? 'selected' : ''}>Start WiFi</option>
+      <option value='6' ${v['action']===6 ? 'selected' : ''}>Enter Binding Mode</option>
+    </select>
+    <label>Action</label>
+  </div>
+</td>
+<td>
+  <div class="mui-select">
+    <select id="select-press-${b}-${p}" onchange="changePress(${b}, ${p}, this.value);">
+      <option value='' disabled hidden ${v['action']===0 ? 'selected' : ''}></option>
+      <option value='false' ${v['is-long-press']===false ? 'selected' : ''}>Short press (click)</option>
+      <option value='true' ${v['is-long-press']===true ? 'selected' : ''}>Long press (hold)</option>
+    </select>
+    <label>Press</label>
+  </div>
+</td>
+<td>
+  <div class="mui-select" id="mui-long-${b}-${p}" style="display:${buttonActions[b][p]['is-long-press'] ? "block": "none"};">
+    <select id="select-long-${b}-${p}" onchange="changeCount(${b}, ${p}, this.value);">
+      <option value='' disabled hidden ${v['action']===0 ? 'selected' : ''}></option>
+      <option value='0' ${v['count']===0 ? 'selected' : ''}>for 0.5 seconds</option>
```

Comment:
The long presses list them as being in 0.5 second increments, but button.cpp works in full second increments, so if I set it to 2 seconds it took ~4 seconds to activate. Also note that the callback reports _repeat count_ not total count so the first long press is `getLongCount() == 0`. The lambda function might just add 1 to getLongCount to get actual seconds?

Half seconds is really weird as an option in my opinion too. Who would set something for 1 second vs 1.5 seconds? Maybe I don't have the reflexes to have things set for half second precision 😄

Reply:
How is it every second when the MS_LONG in button.h is 500!?

Reply:
Maybe your reactions are faster than you think 😄